app/IATI/Services/Audit/AuditService.php

```diff
@@ -190,4 +203,99 @@ public function getNewValues($auditables, $auditableType, $event): bool | string
 
         return json_encode($auditables);
     }
+
+    /**
+     * Returns auditable id.
+     *
+     * @param $event
+     * @return ?int
+     */
+    public function getAuditableId($event): ?int
+    {
+        switch ($event) {
+            case 'signin':
+                return $this->getUserId();
+            case str_contains($event, 'migrated'):
+                return  $this->auditableId;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Sets auditable id.
+     *
+     * @param int $auditableId
+     * @return $this
+     */
+    public function setAuditableId(int $auditableId): static
+    {
+        $this->auditableId = $auditableId;
+
+        return $this;
+    }
+
+    /**
+     * Audit migration event.
+     *
+     * @param $element
+     * @param $event
+     *
+     * @return AuditService
+     */
+    public function auditMigrationEvent($element, $event): static
+    {
+        $row = [];
+        $row['user_type'] = $this->getUserType();
+        $row['user_id'] = $this->getUserId($event);
+        $row['auditable_type'] = get_class($element);
+        $row['auditable_id'] = $this->getAuditableId($event);
+        $row['event'] = $event;
+        $row['url'] = request()->getUri();
+        $row['ip_address'] = request()->getClientIp();
+        $row['user_agent'] = request()->userAgent();
+        $row['created_at'] = now();
+        $row['updated_at'] = now();
+        $row['new_values'] = $this->encryptAuditableItemIfUserOrOrganization($element->toArray(), $row['auditable_type']);
+
+        $this->auditRepository->insertRows($row);
+
+        return  $this;
+    }
+
+    /**
+     * Returns plain text auditableItem or encrypted if auditableItem is User or Org.
+     *
+     * @param array $auditableItem
+     * @param string $auditableType
+     *
+     * @return string|bool
+     */
+    public function encryptAuditableItemIfUserOrOrganization(array $auditableItem, string $auditableType): string|bool
+    {
+        if ($auditableType === get_class(new User) || $auditableType === get_class(new Organization)) {
+            $auditableItem = $this->recursivelyEncrypt($auditableItem);
+        }
+
+        return json_encode($auditableItem);
+    }
+
+    /**
+     * Recursively Encrypt an array.
+     *
+     * @param $item
+     * @return mixed|string
```

Comment:
new line and try to not use mixed
also if mixed is declared. no need to declare string